pyplanet/apps/contrib/brawl_match/__init__.py

```diff
@@ -0,0 +1,247 @@
+
+		await self.brawl_chat(f'Map order: ')
+		for index, (uid, _) in enumerate(self.match_maps, start=1):
+			map_name = (await Map.get_by_uid(uid)).name
+			await self.brawl_chat(f'[{index}/{len(self.match_maps)}] {map_name}')
+
+		await self.set_match_settings()
+
+	async def await_match_start(self):
+		await asyncio.sleep(self.TIME_UNTIL_NEXT_WALL)
+		await self.brawl_chat(f'Match will start in {self.TIME_UNTIL_MATCH_PHASE} seconds!')
+		await asyncio.sleep(self.TIME_UNTIL_BAN_PHASE / 2)
+		await self.brawl_chat(f'Match will start in {int(self.TIME_UNTIL_MATCH_PHASE/2)} seconds!')
+		await asyncio.sleep(self.TIME_UNTIL_BAN_PHASE / 4)
+		await self.brawl_chat(f'Match will start in {int(self.TIME_UNTIL_MATCH_PHASE/4)} seconds!')
+		await asyncio.sleep(self.TIME_UNTIL_BAN_PHASE / 4)
+		await self.brawl_chat(f'Match will start now!')
+		await asyncio.sleep(self.TIME_UNTIL_NEXT_WALL)
+
+
+	async def stop_match(self, player, *args, **kwargs):
+		if not self.match_tasks:
+			await self.brawl_chat(f'No match is currently in progress!', player)
+			return
+		for task in self.match_tasks:
+			if not task.done():
+				task.cancel()
+		self.match_tasks = []
+		await self.brawl_chat(f'Admin {player.nickname}$z$fff stopped match!')
+		for signal, target in self.context.signals.listeners:
+			if target == self.set_settings_next_map or target == self.set_settings:
+				signal.unregister(target)
+		self.match_maps = self.brawl_maps.copy()
+		self.match_players = []
+
+		await self.reset_backup()
+
+	async def reset_backup(self):
+		if not (self.backup_script_name and self.backup_settings):
+			return
+		await self.instance.mode_manager.set_next_script(self.backup_script_name)
+		await self.instance.gbx('RestartMap')
+		await self.instance.mode_manager.update_next_settings(self.backup_settings)
+
+
+	async def update_finish_timeout(self, timeout):
+		settings = await self.instance.mode_manager.get_settings()
+		settings['S_FinishTimeout'] = timeout
+		await self.instance.mode_manager.update_settings(settings)
+
+	async def set_settings_next_map(self, map):
+		settings = await self.instance.mode_manager.get_settings()
+		for index, (uid, timeout) in enumerate(self.match_maps):
+			if uid == map.uid:
+				if index == len(self.match_maps) - 1 and settings['S_WarmUpNb'] == 1:
+					settings['S_WarmUpNb'] = 0
+					await self.instance.mode_manager.update_settings(settings)
+				await self.update_finish_timeout(timeout)
+				await self.instance.map_manager.set_next_map(
+					await Map.get_by_uid(
+						self.match_maps[(index + 1) % len(self.match_maps)][0]
+					)
+				)
+
+	async def brawl_chat(self, message, player=None):
+		if player:
+			await self.instance.chat(f'{self.chat_prefix}{message}', player)
+		else:
+			await self.instance.chat(f'{self.chat_prefix}{message}')
+
+	async def register_match_task(self, func, *args):
+		if args:
+			self.match_tasks.append(asyncio.create_task(func(*args)))
+		else:
+			self.match_tasks.append(asyncio.create_task(func()))
```

Comment:
Is this check needed?I think this will still work if you just have `self.match_tasks.append(asyncio.create_task(func(*args)))`